clans/src/main/java/me/mykindos/betterpvp/clans/clans/ClanManager.java

```diff
@@ -152,6 +153,37 @@ public boolean hasAccess(Player player, Location location) {
         return relation == ClanRelation.SELF || relation == ClanRelation.ALLY_TRUST;
     }
 
+    public Location closestWilderness(Player player) {
+        List<Location> locations = new ArrayList<>();
+        //worst case scenario, we are stuck in the exact center of a 3x3 claim
+        for (int i = 0; i < ((16 + 8) + 1); i++) {
+            for (int d = 0; d < 4; d++) {
+                Location location = player.getLocation().toBlockLocation();
+                switch (d) {
+                    case 0:
+                        location.add(i, 0, 0);
+                        break;
+                    case 1:
+                        location.add(0, 0, i);
+                        break;
+                    case 2:
+                        location.add(-i, 0, 0);
+                        break;
+                    case 3:
+                        location.add(0, 0, -i);
+                }
+                Optional<Clan> clanOptional = getClanByLocation(location);
+                if (clanOptional.isEmpty()) {
+                    locations.add(location);
+                }
+            }
+        }
+        if (!locations.isEmpty()) {
+            locations.sort(Comparator.comparingInt(a -> (int) player.getLocation().distance(a)));
```

Comment:
If people were to spam the `StuckSubCommand` in a claim in which they're not able to teleport out of (greater than the area you scan), this could cause performance issues not only by the sort method call but also by the distance calculation you make for each possible location, which seems to be on a per-block basis based on this implementation.

Reply:
I'll make a small change to the DelayedAction so it does nothing if the player already has one pending which will help a little.

A better solution here might be to get a list of your surrounding chunks, sort it by closest chunk, then find the best location within that chunk only to teleport the player to.

Could also look to implement a small cooldown (5s or so) on the command

Reply:
Implemented, please check.